upgrade/migrate.go

```diff
@@ -79,3 +90,43 @@ func migrateYAMLToJSON() error {
 
 	return nil
 }
+
+func migrateNoNameToName() error {
```

Comment:
This seems to assign `general_note_0` to all notes as names.

problems:

1. book name in the name is wrong
2. note number is not incremental - it is because this function does not save the content to `.dnote` until the very end. But `utils.GenerateNoteName` reads `.dnote` every time to determine the next number.

We can save to `.dnote` inside the loop in this method so that `utils.GenerateNoteName` knows the next number. But that seems kind of inefficient. Suggestions welcomed.